### DISCUSSION TITLE
On demand question

### DISCUSSION THREAD

**Discussion by dnestico** [mrlt8/docker-wyze-bridge#499] [Q&A]
So I enabled on demand streams from the new update and the cameras are supposed to appears as "unavailable" when they're not being viewed right? Or is it broken?

Also it doesn't seem to really work, I have it combined with he WebRTC card and it just doesn't work a lot of the time. I even turned it OFF but somehow in the logs it was still loading it as if the on demand setting was still on when it was OFF...

**Comment by mrlt8** [comment]
The WEBRTC option is for pulling the credentials for the WebRTC stream (the stream that the web view, Alexa, etc use).

The ON_DEMAND option will prevent the bridge from connecting to the cameras until someone tries to access the streams on the bridge. Unfortunately, it does take a couple of seconds to connect to the cams and pull the stream which can cause some clients to time out. 

You should see a yellow clock icon on the cameras that are "on-demand" in the web-ui.

Edit: Filters still apply with on-demand, so if you have some camera filtered out, it will still be ignored.

**Comment by dnestico** [comment]
By webRTC I mean I am using the webRTC card haha aka this https://github.com/AlexxIT/WebRTC, the credentials thing hasn't really even worked for me tbh. 

And yes I see the clock, it just seems the cameras do timeout the first time and I get no video from them, also the thumbnails are old which makes sense I guess so I'ma try this for now and see if its something I'll stick with tbh.

**Reply by mrlt8** [reply]
v1.8.2 should fix the timeout issue.

If you want to refresh the images, you'll have to click the refresh icon that pops up over the image.

**Reply by eloquentvogon** [reply]
can we schedule updates for the on demand cams and save that picture local to the server somewhere?

**Reply by mrlt8** [reply]
Yes, `SNAPSHOT=RTSP` should continue to work even when the cameras are on-demand. 

You can adjust the frequency with something like `SNAPSHOT=RTSP600` to take a snapshot every 600 seconds.

The snapshot will be saved to /img/ inside the container, so you'll need to mount that to your local system or pull the image from the WebUI.

If you're using docker-compose you can do something like:

```yaml
    - SNAPSHOT=RTSP600
volumes:
    - /path/to/save/local/img/:/img/
```

**Reply by eloquentvogon** [reply]
interesting, I don't think that is working on mine, I keep getting a 304 message on query. I will have to check my config, thanks

**Reply by eloquentvogon** [reply]
I increased my RTSP to 600 from 60, seemed to work initially but then started giving 304 again, I have logs if you want them

**Reply by mrlt8** [reply]
Hmm, it could be timing out if the on-demand camera takes too long to start up...
Another way would be to trigger a new snapshot using the /snapshot/cam-name.jpg endpoint from the webui.

**Reply by eloquentvogon** [reply]
I'm trying to update the snapshot for a home assistant dashboard, as the outdoors do not like constant streaming. is there anyway to initiate that req from HA, or is there cron capability in the container to make that request?

BTW thanks for doing this project, it makes wyze a usable platform

**Reply by eloquentvogon** [reply]
I have HA grabbing the snapshot URL on refresh, thanks! there delay was throwing me off

**Reply by mrlt8** [reply]
Sorry, my bad! SNAPSHOT=RTSP does NOT work with on demand. 
I completely forgot that it's only triggered _after_ a camera goes up.

**Reply by eloquentvogon** [reply]
Ah ok. Thanks for that !

**Comment by dnestico** [comment]
Sweet! I only got 1.8.1 so hopefully tomorrow I get 1.8.2

**Reply by dnestico** [reply]
And I litterally just got 1.8.2 after updating to 1.8.1 lol.

**Comment by backslashV** [comment]
Question about the on-demand option. I have added _- platform: ffmpeg_ cameras with the rtsp URL of them (from this addon) to HA. These cameras are then added to a picture-entity card on the UI.
Will accessing the UI page containing these picture-entities cause the stream to begin and once you navigate away from the page, they will stop streaming?

**Reply by mrlt8** [reply]
It _should_, but I'm not sure how often home assistant attempts to ping the streams in the background. 

I did try playing around with it and noticed it trying to pull the stream even when I wasn't viewing the stream in HA. May need to make some adjustments in the HA config to only pull the stream when viewing.
